app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarLayoutViewModel.kt

```diff
@@ -131,7 +132,7 @@ class OmnibarLayoutViewModel @Inject constructor(
                 (state.viewMode is NewTab || state.hasFocus && state.omnibarText.isNotBlank() || duckChat.isEnabledInBrowser()),
             showClickCatcher = isDuckAIPoCEnabled,
         )
-    }.flowOn(dispatcherProvider.io()).stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000L), _viewState.value)
```

Comment:
Do you know why this was 5000ms before? Is it safe to change?

Reply:
I had a comment drafted for this which I didn't submit :facepalm: Here goes:

I'm making this flow eager because the many direct value checks in this class can behave unpredictably if the flow doesn’t emit values, potentially causing subtle bugs. In practice, the flow is always observed by the omnibar, so this change shouldn't impact behavior but removes that risk and simplifies unit testing.

I spoke with @0nko some time ago and the timeout was added as a resource-saving precaution in case the flow wasn’t consumed. But since in our case it's always expected to be consumed, and the internal logic depends on emitted values, I think it's safer to make that explicit by using an eager flow. @0nko please keep me honest about it though in case there's a reason to keep the flow lazy.

Reply:
@LukasPaczos That's right, the original reason still holds, I've considered `SharingStarted.WhileSubscribed(5000L)` sort of the default to saved resources based on some tutorials when I first started using the flows, unless there was a reason to change it, which it seems, we do have. So all good 👍.